executor/src/witgen/machines/block_machine.rs

```diff
@@ -315,167 +331,66 @@ impl<'a, T: FieldElement> BlockMachine<'a, T> {
             }
         }
 
-        let old_len = self.rows();
-        self.append_new_block(fixed_data.degree)?;
-        let mut outer_assignments = EvalValue::complete(vec![]);
-
-        // While processing the block, we'll add an additional row which will be
-        // removed once the block is done. This is done to prevent infinite loops:
-        // If we ignored updates to the last row, they would be computed over
-        // and over again.
-        self.data.push(self.row_factory.fresh_row());
-
         // TODO this assumes we are always using the same lookup for this machine.
-        let mut processing_sequence_iterator =
-            self.processing_sequence_cache.get_processing_sequence(left);
-
-        let mut errors = vec![];
-        // TODO The error handling currently does not handle contradictions properly.
-        // If we can find an assignment of all LHS variables at the end, we do not return an error,
-        // even if there is a conflict.
-
-        // A copy of `left` which is mutated by `handle_outer_constraints()`
-        let mut left_mut = left.to_vec();
-
-        // Can't use a for loop here, because we need to communicate progress_in_last_step to the
-        // default iterator.
-        while let Some(step) = processing_sequence_iterator.next() {
-            let SequenceStep {
-                row_delta,
-                identity,
-            } = step;
-            let row = (old_len as i64 + row_delta + fixed_data.degree as i64) as DegreeType
-                % fixed_data.degree;
-
-            let progress = match self.compute_updates(
-                row,
-                fixed_data,
-                &left_mut,
-                right,
-                identity,
-                &mut identity_processor,
-            ) {
-                Ok(value) => {
-                    if !value.is_empty() {
-                        errors.clear();
-
-                        let progress = self.apply_updates(row, identity, &value, &mut left_mut);
-
-                        for (poly, constraint) in value.constraints {
-                            if !self.witness_cols.contains(&poly.poly_id()) {
-                                outer_assignments.constraints.push((poly, constraint));
-                            }
-                        }
-
-                        progress
-                    } else {
-                        false
-                    }
-                }
-                Err(e) => {
-                    errors.push(format!("In row {}: {e}", row).into());
```

Comment:
The error handling is different now. I don't quite understand why the old behavior was the way it was: Why do we need to collect multiple errors? Why are `EvalError`s ignored if the next step succeeds?).

Now, any `EvalError` that occurs within `Processor::solve()` will be bubbled up directly to the caller via the `?` operator.